### PR TITLE
fix: add more columns to ledger entries query result

### LINKED ISSUE #6808: LedgerEntries API returns data that is missing account_type and transfer_type
## Problem encountered

Whilst testing and integrating with the v2 LedgerEntries API on Stagnet1RC I observed some strangeness where we see that the LedgerEntries API returns data that is missing account_type and transfer_type for a party as shown by the query below:

https://api.n00.stagnet1.vega.xyz/api/v2/ledgerentry/history?filter.accountFromFilter.partyIds=d8742dbe34198241b96381012be71615a48da2b1c60f76f25ae5ed87fea9c500

```
Ledger entries:
{
  "ledgerEntries": {
    "edges": [
      {
        "cursor": "eyJ2ZWdhX3RpbWUiOiIyMDIyLTExLTExVDE0OjE2OjA4Ljc0MjcxNloifQ==",
        "node": {
          "accountType": "ACCOUNT_TYPE_UNSPECIFIED",
          "quantity": "10000000000000000000000",
          "timestamp": "1668176168742716000",
          "transferType": "TRANSFER_TYPE_UNSPECIFIED"
        }
      }
    ],
    "pageInfo": {
      "endCursor": "eyJ2ZWdhX3RpbWUiOiIyMDIyLTExLTExVDE0OjE2OjA4Ljc0MjcxNloifQ==",
      "hasNextPage": false,
      "hasPreviousPage": false,
      "startCursor": "eyJ2ZWdhX3RpbWUiOiIyMDIyLTExLTExVDE0OjE2OjA4Ljc0MjcxNloifQ=="
    }
  }
}
```

## Expected behaviour

As a public API user I'd expect there to be more information about what this ledger movement was for (types). It doesnt tell the user what accountType or what transferType it was! I'd have expected something like transfer type deposit, account type general?

## Further digging

Investigating that party shows that the party DID deposit that amount (10000) of VEGA (stagnet 1) token and it is status is Finalized however the only transfer they have is a failed attempt to send (1000) VEGA (stagnet 1) token to the global reward pool.

### LINKED ISSUE #6809: LedgerEntries API queries can be slower than expected
We have observed that some of the queries run against ledger entries can be slow (minutes to return) on the v2 LedgerEntries API. Please see an example (Jeremy saw this in particular):

```
query LE($filter: LedgerEntryFilter) {
	ledgerEntries(filter: $filter) {
  edges {
    node {
      vegaTime
      quantity
      accountType
      marketId
      assetId
      transferType
    }
  }
  }
}
```

### PR DESCRIPTION
**Pull request by AnExsomnis**
close: #6808 
close: #6809

### CODE REVIEW

File: datanode/api/trading_data_v2.go [hunk 1]
```diff
@@ -220,39 +220,13 @@ func (t *tradingDataServiceV2) ListLedgerEntries(ctx context.Context, req *v2.Li
...
 	dateRange := entities.DateRangeFromProto(req.DateRange)
 	pagination, err := entities.CursorPaginationFromProto(req.Pagination)
 	if err != nil {
 		return nil, apiError(codes.InvalidArgument, fmt.Errorf("invalid cursor: %w", err))
 	}
 
-	entries, pageInfo, err := t.ledgerService.Query(leFilter, groupOptions, dateRange, pagination)
+	entries, pageInfo, err := t.ledgerService.Query(leFilter, dateRange, pagination)
 	if err != nil {
 		return nil, fmt.Errorf("could not query ledger entries: %w", err)
```
Comment (jeremyletang) [inline datanode/api/trading_data_v2.go:231]: Could this return an apiError instead?

**Review by jeremyletang — COMMENTED** [review]
looks good, but would be good to have @guoguojin and @pscott31 to look into it before we merge.

File: datanode/entities/aggregated_ledgerentries.go [hunk 2]
```diff
@@ -41,23 +45,41 @@ func (ledgerEntries *AggregatedLedgerEntries) ToProto() *v2.AggregatedLedgerEntr
...
+			lep.PartyId = &partyIDString
+		}
+	}
+
+	if ledgerEntries.AccountToPartyID != nil {
+		partyIDString := ledgerEntries.AccountToPartyID.String()
+		if partyIDString != "" {
+			lep.PartyId = &partyIDString
+		}
+	}
```
Comment (pscott31) [inline datanode/entities/aggregated_ledgerentries.go:60]: is it intended that the toParty overrides the fromParty here? I might have gotten the wrong end of the stick but I kind of expected a proto change to add FromPartyId, ToPartyID etc..

File: datanode/api/trading_data_v2.go (hunk 1 above)
Comment (AnExsomnis) [reply datanode/api/trading_data_v2.go]: Addressed here https://github.com/vegaprotocol/vega/pull/6834/files#diff-a493a31fca9e62f51a5f1c9971fa04a87210e5ff399f4737d5f8a610c5eb17c8R230
Thanks!

File: datanode/entities/aggregated_ledgerentries.go [hunk 3]
```diff
@@ -18,20 +18,24 @@ type AggregatedLedgerEntries struct {
...
 	AssetID      *AssetID
-	MarketID     *MarketID
-	AccountType  *types.AccountType
+
+	AccountFromPartyID     *PartyID
+	AccountFromMarketID    *MarketID
+	AccountFromAccountType *types.AccountType
+	AccountToPartyID       *PartyID
+	AccountToMarketID      *MarketID
+	AccountToAccountType   *types.AccountType
```
Comment (pscott31) [inline datanode/entities/aggregated_ledgerentries.go:28]: Super minor and a matter of opinion (e.g. non merge blocking) but i'd probably have just called these e.g. `FromPartyID`

**Review by pscott31 — COMMENTED** [review]
I feel like there are some missing proto changes to go with this (e.g. at least removal of groupOptions?) and maybe separate fromParty/toParty etc.. [though that might not be part of the plan, just seems odd to go to the effort of querying them not to return them though]

File: datanode/entities/aggregated_ledgerentries.go (hunk 2 above)
Comment (AnExsomnis) [reply datanode/entities/aggregated_ledgerentries.go:60]: No, I forgot about that.
Sorry, will add it now

Comment (AnExsomnis) [reply datanode/entities/aggregated_ledgerentries.go]: @pscott31 , addressed this:
Removed groupoptions from protos and gqlschema, as well as added the new fields:
https://github.com/vegaprotocol/vega/pull/6834/files#diff-7f6c5ea53fa01231a8a36ffa313993450073ba666c91f11fe0322f33a39b1b86
https://github.com/vegaprotocol/vega/pull/6834/files#diff-be016f833f94c6069e96beb4873414c55bdfb7af1924c9c1b90eb709304ff90e

Comment (AnExsomnis) [reply datanode/entities/aggregated_ledgerentries.go]: Also corrected the serialization: https://github.com/vegaprotocol/vega/pull/6834/files#diff-1b82afd4970ab61dfe18dca2347cc372dee6b86434f92c37d975daee780796afL27

File: datanode/entities/aggregated_ledgerentries.go (hunk 3 above)
Comment (AnExsomnis) [reply datanode/entities/aggregated_ledgerentries.go:28]: Do you mean `FromPartyId`, `ToPartyId`, `FromMarketId`, `ToMarketId`, `FromAccountType`, `ToAccountType`? 
If so, what about 
`SenderPartyId`, `ReceiverPartyId`, `SenderMarketId`, `ReceiverMarketId`?

Comment (pscott31) [reply datanode/entities/aggregated_ledgerentries.go:28]: I like that

Comment (AnExsomnis) [reply datanode/entities/aggregated_ledgerentries.go:28]: All right, should i include that change now?

Comment (AnExsomnis) [reply datanode/entities/aggregated_ledgerentries.go:28]: it makes sense to do so

Comment (AnExsomnis) [reply datanode/entities/aggregated_ledgerentries.go]: All right the change is applied

**Review by pscott31 — DISMISSED** [review]

**Review by jeremyletang — DISMISSED** [review]

**Review by jeremyletang — DISMISSED** [review]

File: protos/sources/data-node/api/v2/trading_data.proto [hunk 4]
```diff
@@ -715,30 +715,30 @@ message AggregatedLedgerEntries {
...
-    // If grouping by transfer type, the transfer type
-    vega.TransferType transfer_type = 7;
-    // If grouping by account type, the acount type
-    vega.AccountType account_type = 8;
-}
-
-message GroupOptions {
-    repeated AccountField by_account_field = 1;
-    repeated LedgerEntryField by_ledger_entry_field = 2;
+    // The account sender acount type
```
Comment (candida-d) [inline protos/sources/data-node/api/v2/trading_data.proto:722]:
```suggestion
    // The type of account sent from
```

File: protos/sources/data-node/api/v2/trading_data.proto [hunk 5]
```diff
@@ -715,30 +715,30 @@ message AggregatedLedgerEntries {
...
+    vega.AccountType sender_account_type = 6;
+    // The account receiver acount type
```
Comment (candida-d) [inline protos/sources/data-node/api/v2/trading_data.proto:724]:
```suggestion
    // The type of account received to
```

File: protos/sources/data-node/api/v2/trading_data.proto [hunk 6]
```diff
@@ -715,30 +715,30 @@ message AggregatedLedgerEntries {
...
+    vega.AccountType receiver_account_type = 7;
+    // If filtering by sender party, the party ID
+    optional string sender_party_id = 8;
+    // If filtering by receiver party, the party ID
+    optional string receiver_party_id = 9;
+    // If filtering by sender market, the market ID
+    optional string sender_market_id = 10;
+    // If filtering by market  market, the market ID
```
Comment (candida-d) [inline protos/sources/data-node/api/v2/trading_data.proto:732]:
```suggestion
    // If filtering by receiver market, the market ID
```

**Review by candida-d — DISMISSED** [review]

**Review by candida-d — APPROVED** [review]

**Review by pscott31 — APPROVED** [review]

**Pull request merged**